### DISCUSSION TITLE
FAQ of YOLO-World

### DISCUSSION THREAD

**Discussion by wondervictor** [AILab-CVC/YOLO-World#149] [Q&A] [answered]
Welcome everyone's attention to YOLO-World ❤️. I hope everyone will share the **problems (or bugs🐛)** encountered or **solutions 🔧** in this discussion. This will facilitate the maintenance of the YOLO-World community, promote everyone's communication 💬 , and further enhance and optimize the YOLO-World model 👊.
> 欢迎大家关注YOLO-World ❤️。希望大家能在这个讨论中分享遇到的问题（或者错误🐛）以及解决方案 🔧。这将有助于维护YOLO-World社区，促进大家的交流 💬 ，并进一步提升和优化YOLO-World模型 👊。

**Comment by xiyangyang99** [comment]
> Welcome everyone's attention to YOLO-World ❤️. I hope everyone will share the **problems (or bugs🐛)** encountered or **solutions 🔧** in this discussion. This will facilitate the maintenance of the YOLO-World community, promote everyone's communication 💬 , and further enhance and optimize the YOLO-World model 👊.
> 
> > 欢迎大家关注YOLO-World ❤️。希望大家能在这个讨论中分享遇到的问题（或者错误🐛）以及解决方案 🔧。这将有助于维护YOLO-World社区，促进大家的交流 💬 ，并进一步提升和优化YOLO-World模型 👊。

这是昨天更新之后，导出onnx的问题；
Traceback (most recent call last):
  File "/media/bowen/6202c499-4f0a-4280-af7e-d2ab4b6c74dd/home/bowen/yolo-world-v2/YOLO-World-master/deploy/easydeploy/tools/export_onnx.py", line 157, in <module>
    main()
  File "/media/bowen/6202c499-4f0a-4280-af7e-d2ab4b6c74dd/home/bowen/yolo-world-v2/YOLO-World-master/deploy/easydeploy/tools/export_onnx.py", line 116, in main
    deploy_model(fake_input)
  File "/home/bowen/anaconda3/envs/mmdet/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/media/bowen/6202c499-4f0a-4280-af7e-d2ab4b6c74dd/home/bowen/yolo-world-v2/YOLO-World-master/deploy/easydeploy/model/model.py", line 166, in forward
    neck_outputs = self.baseModel(inputs)
  File "/home/bowen/anaconda3/envs/mmdet/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/media/bowen/6202c499-4f0a-4280-af7e-d2ab4b6c74dd/home/bowen/mmdetection-3.0.0/mmdet/models/detectors/base.py", line 96, in forward
    return self._forward(inputs, data_samples)
  File "/media/bowen/6202c499-4f0a-4280-af7e-d2ab4b6c74dd/home/bowen/yolo-world-v2/YOLO-World-master/yolo_world/models/detectors/yolo_world.py", line 70, in _forward
    img_feats, txt_feats = self.extract_feat(batch_inputs,
  File "/media/bowen/6202c499-4f0a-4280-af7e-d2ab4b6c74dd/home/bowen/yolo-world-v2/YOLO-World-master/yolo_world/models/detectors/yolo_world.py", line 81, in extract_feat
    texts = self.texts
  File "/home/bowen/anaconda3/envs/mmdet/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1614, in __getattr__
    raise AttributeError("'{}' object has no attribute '{}'".format(
AttributeError: 'YOLOWorldDetector' object has no attribute 'texts'


导出的时候，继承的类没有text这个参数，官方pth和fintune的pth都是这个问题。

**Reply by wondervictor** [reply, marked as answer]
refer to [`docs/deploy`](https://github.com/AILab-CVC/YOLO-World/blob/master/docs/deploy.md) and try [`deploy/export_onnx`](https://github.com/AILab-CVC/YOLO-World/blob/master/deploy/export_onnx.py).

**Comment by yinchunlin** [comment]
模型[YOLO-Worldv2-X]的官方权重在哪里下载呀？

**Comment by yinchunlin** [comment]
请问，要下载官方的[YOLO-Worldv2-X]预训练模型权重在哪里下载？

**Reply by wondervictor** [reply]
可以查看首页README: https://github.com/AILab-CVC/YOLO-World?tab=readme-ov-file#zero-shot-inference-on-lvis-dataset
后面附有HF链接

**Comment by karlcheneric** [comment]
请问，作为一个新人，我有传统的yolo训练所用的图像数据集即照片.jpg文件，请问我应该用什么工具才能创建Yolo-world训练所用的图像-文本对数据集呢

**Reply by wondervictor** [reply]
@karlcheneric 现在是只有图像没有文本数据？

**Reply by karlcheneric** [reply]
我有图像数据和对应的文本数据，但是不清楚如何将他们组合在一起，比如说使用什么工具呢？我有使用labelme创建.json文件作为标注的经历，但是我不知道怎么创建与图像匹配的文本数据对。能否请您帮助我建立新的数据集用于训练呢

**Reply by wondervictor** [reply]
可以采用YOLO-World-X-1280对数据进行伪标注，可以先从文本中提取noun phrases，直接用X-1280模型进行预测，并过滤筛选得到标注。

**Comment by joy-seu** [comment]
请问 有导出的onnx模型吗？

**Reply by wondervictor** [reply]
是遇到什么问题了吗

**Comment by Cynthiahahahah** [comment]
你好，我有之前在用yolov5和yolov8训练时进行标注的自定义数据集，可用于训练yolov5和v8。现在我想要使用该自定义数据集对yolo-world进行训练，我该怎么去做、使用什么工具呢（比如用什么工具去标注图片、怎么得到annotation.json和class_texts.json
，怎么去建立图片和文本之间的联系等等）？作为一名新人，希望您回答的尽量详细一些，谢谢~

**Comment by chengguixiong1010** [comment]
yolo-world-v2-seg模型权重在哪可以下载呢?

**Reply by wondervictor** [reply]
预计10号之前一并上传。

**Reply by chengguixiong1010** [reply]
有上传吗? 我怎么没有找到呢

**Comment by guanning03** [comment]
请问如何理解这些checkpoints的命名含义？如果微调的话用哪个最好？
![image](https://github.com/user-attachments/assets/ceea944b-4f74-4a8d-b9d6-cd8b428f5d11)

**Reply by guanning03** [reply]
或者说，哪个是比较标准的？

**Comment by kill7893** [comment]
你好，请问为什么我加载了提供的预训练权重文件，去做视频预测，预测的效果很差呢？还是说提供的预训练权重是没有经过训练的呢？

**Reply by kill7893** [reply]
使用的是这个：YOLO-Worldv2-XL

**Comment by yunfei-ai** [comment]
嗨，我这里遇到了一个问题，是训练自定义数据集，改变了类别数，导致index error
![Uploading 屏幕截图 2025-04-03 223544.png…]()

**Comment by ihaveaword** [comment]
# 在线演示好像出问题了
<img width="1103" height="570" alt="image" src="https://github.com/user-attachments/assets/bf5e606c-a79f-42c2-90f8-f6740370863d" />

**Comment by SZhaoJiaXing** [comment]
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Cell In[43], line 1
----> 1 from ultralytics import YOLOWorld

ImportError: cannot import name 'YOLOWorld' from 'ultralytics' (D:\conda\Lib\site-packages\ultralytics\__init__.py)

I met this problem ,and I use pip install -U ultralytics  but also met this proble  , how to deal with it ?

**Comment by tianle-BigRice** [comment]
麻烦给出一个可以正常配置的环境配置把。目前给出的方法去配置环境都无法配置成功，各种环境冲突
